### ISSUE TITLE
Button "Create New" in entries' associations collides with title

### ISSUE THREAD

**Issue by mprofitl** [symphonycms/symphony-2#2650]
The sidebar `associations` show a title of each associated section and right beneath a button `Create New`. The sidebar is not wide enough for showing them beneath each other. I suggest to change the css of `symphony.associations.css` to 
```
.association-new {
   margin-top: 1rem;
   margin-left:0;
}
```
Then the button will be shown underneath the title without collision.

**Comment by twiro** [comment]
> _The sidebar is not wide enough for showing them beneath each other._

That surely depends on the length oh your section-names ;)

> _I suggest to change the css of symphony.associations.css ... Then the button will be shown underneath the title without collision._

I don't think this will be a more satisfying solution than what we currently have. If you keep your section-names rather short, but have a lot of association-stuff going on in the drawer (and I often have lots of) you're glad that the "create new"-buttons don't take away any extra vertical space.

On top of that putting the buttons below the section-names doesn't add to the clearness/readability of the drawer.vI've attached a quick comparison-screenshot with a rather full associations-drawer to show what I mean.

One possible solution might be to simply shorte the section-titles if they're longer than a maximum of allowed characters. Or giving the drawer itself a little more width... or best combining both these methods. At least the Buttons should be placed above the section-title (z-index-wise)...

![symphony-associations-drawer](https://cloud.githubusercontent.com/assets/870227/23671917/45df1bc0-036d-11e7-8fac-b585e085237a.jpg)

**Comment by mprofitl** [comment]
Your screenshots help a lot – I am adding one now.

Once a association exists, the collision rarely happens. However when no association exists yet, the text (at least with the backend in German langauge) is too long for text and button, e.g. _Keine Verknüpfungen für Neuigkeiten_ and _Neu erstellen_.

The increased z-index will make the button usable no mater how long the text. That would be a good amendment. 

The button on the right is indeed visually much clearer. The length of the section names don't seem to be troublesome, comparing yours and mine, it seems to be the translation. On the image attached I tried alternatives for the naming:

- _Anlegen_, _Erstellen_ or _Neu_ instead of _Neu erstellen_
- No differentation between the stated links existant or not: in both cases just _Verknüpfungen für…_ or even _Links für…_

Shortening the section titles would not help a lot as the space left to the button is already used up almost by the label.

![symphony associations in drawer](https://cloud.githubusercontent.com/assets/6146601/23692038/8fd02d00-03cc-11e7-9872-1ff7a29cce2b.png)

**Comment by twiro** [comment]
Thanks for your further explanation! And you're right - your section naming isn't to blame here, it's the bloody german language ;) That's why I always avoid it in the backend - it's simply too verbose for interface pruposes!

Nonetheless the styling has a bug that can be fixed and I tried to come up with [a proper solution](https://github.com/symphonycms/symphony-2/compare/master...twiro:association-drawer-create-new-styling-fix) that avoids ugly overlapping when the association title is getting too long - Apart from simply adding a proper z-index to the button I also included a pseude-element (highlighted in the second example) with a small gradient that makes the overlapping a little less ugly (visible in the left example):

![symphony-associations-drawer-2](https://cloud.githubusercontent.com/assets/870227/23938791/15c8af36-095e-11e7-889c-fa3e2e9c16f3.jpg)

Let me know if that helps and what you think of it! If @nitriques is fine with this (after returning from his paternal leave) I will send a PR!

----

Regarding the wording you best have a small chat with @animaux as he's the maintainer of the german translation. And maybe @nilshoerrmann also has some thoughts about this...

As I assume that the string **"Neu erstellen"** is the same one that's used in the super-important green button on the section-index-pages, therefore I'd be careful with any changes!

But regarding only the association drawer simply using **"Erstellen"** as a short form of **"Neu erstellen"** _(that might be a better choice for the default green button simply due to it's length)_ makes sense to me - especially if the preceding title contains the word "Verknüpfungen" - so I actually like the second solution in the following screenshot that uses the same wording **"Verknüpfungen in XY"** in combination with a simple **"Erstellen"**-Button for both existing and non-exisiting associations:

![symphony-associations-drawer-3](https://cloud.githubusercontent.com/assets/870227/23938972/f252426e-095e-11e7-9044-35693a624ec7.jpg)

But the left one - keeping everything as is except the button wording - would be an improvement too!

**Comment by nilshoerrmann** [comment]
Some thoughts:

I don't remember why we opted for shortened, single line links for this interface, but redesigning this layout with text spreading multiple lines might fix some usability issues. Same is true for the titles that collide with the button.

- The title could break to a new line, when it becomes to long.
- Flexbox might make handling this layout easier (it wasn't available back then).
- There could always be a title bar only consiting of the section name and the button. This would reduce the needed space but would still fail on longer section names as long as there isn't any special treatment (like text shortening, fading or breaking).
- All statical information could be moved to its own line

        Projekte            [Neu erstellen]
        Keine Verknüpfungen

---

I wouldn't touch the translations as that might open Pandora's box.

**Comment by twiro** [comment]
> _I don't remember why we opted for shortened, single line links for this interface, but redesigning this layout with text spreading multiple lines might fix some usability issues. Same is true for the titles that collide with the button._

In my eyes opting for strictly single line links was a well made decision - multiple lines per link would clutter the layout of the drawer even more and none of the cut off title information seems that valuable to me that I would choose "full readability" over having a halfway nice vertical rhythm in the drawer. Plus - as stated in my initial post - vertical space is a precious thing in the drawer and I'd always prefer some long titles getting cut off over having to scroll in the drawer.

> _The title could break to a new line, when it becomes to long._

This would mean (neraly) every association title in a german language backend that has no associations yet would require two lines... I wouldn't like that, but as I run nearly every project with a strictly english backend I wouldn't mind - could be a valid compromise.

> _There could always be a title bar only consiting of the section name and the button. This would reduce the needed space but would still fail on longer section names as long as there isn't any special treatment (like text shortening, fading or breaking).... All statical information could be moved to its own line_

Though this sounds like a clean solution it would also mean loosing additional vertical space with each linked section - would be ok if we kept the actual links in single lines and maybe make the link listings a little tighter... there's quite some space above each single link.

But I think the position of the "Create"-button feels somehow wrong next to the bare section title -  it would be more intuitive on the same line with "XY associations in ..." because that would imply "click this button to create a new **associated project**" rather than "click this button to create a new (whatever) **project**"

    Projekte              [ Neu erstellen ]      Projekte
    ---------------------------------------      ---------------------------------------
    2 Verknüpfungen                              2 Verknüpfungen       [ Neu erstellen ]
    ---------------------------------------      ---------------------------------------
    -> Projektname XYZ                           -> Projektname XYZ
    -> Projektname 123                           -> Projektname 123
    ---------------------------------------      ---------------------------------------

Thinking about it I would prefer a simple one-liner with a **very** short [+]-button, but we don't have something like this in Symphony yet...


    2 Verknüpfungen in Projekte         [+]
    ---------------------------------------
    -> Projektname XYZ
    -> Projektname 123 
    ---------------------------------------

**Comment by nilshoerrmann** [comment]
I like this version:

    Projekte
    ---------------------------------------
    Verknüpfungen       [ Neu erstellen ]
    ---------------------------------------
    -> Projektname XYZ
    -> Projektname 123
    ---------------------------------------

Though I have to admit that I've never ever used the association drawer in a real world project.

**Comment by mprofitl** [comment]
I can't cope with your speed – being busy with architecture (as in houses, not systems) during daytime. I appreciate you are taking time to discuss the drawer.

_Neu erstellen_ is used as the standard wording for all main CTA buttons in Symphony. Thus I agree it would be bad to change it. The buttons in the drawer however aren't main CTAs; the main CTA is _Eintrag erstellen_ or _Änderungen speichern_ (by the way, this button should be green, I believe, and all other main buttons in the backend). All buttons in the drawer are secondary buttons thus it may have a different name in my eyes. I second Roman's idea to drastically shorten it to a [+].

The drawer always shows links, links from and to, doesn't it? Thus repeating the title _Verknüpft mit_ is redundant. What about giving the whole drawer two headers _Verknüpft zu_ and _Verknüpfungen in_ and then just listing associated setions and entries? Those two titles, by the way, appear clearer to me than today _Verknüpft mit_, _Keine Verknüpfungen für_, _Verknüpft in_ (why three names when it's technically just two ways of linking, from and to?).

Keeping the drawer slim seems to me a favourable idea. Placing the [ + ] Button in the same line as the title works towards this aim. As would shortening the amount of links to a number in brackets just behind the title instead of investing a whole line for it. What do you think about this (left with links, right with no links)?

```Verknüpft zu

Gebäudearten (2)                  [ + ]      Gebäudearten (0)                  [ + ]
---------------------------------------      ---------------------------------------
-> Mehrfamilienhäuser                        Keine
-> Bürobauten
---------------------------------------


Verknüpfungen in

Projekte (2)                      [ + ]      Projekte (0)                      [ + ]
---------------------------------------      ---------------------------------------
-> Projektname XYZ                           Keine
-> Projektname 123
---------------------------------------
```
By the way, I use the drawer a lot. For editing I often use the links in the entries list, when creating and entry I use the drawer to add for example images.

**Comment by nitriques** [comment]
Why guys, I am late to the party, but I like this a lot. @twiro Please send a PR!

**Comment by nitriques** [comment]
But you might wanna wait until https://github.com/symphonists/documenter/issues/25 gets fixed...

**Comment by nitriques** [comment]
@twiro Ping! The mentioned issue got fixed, so I would accept a PR based on the latest `2.7.x` branch! Thanks.

**Comment by twiro** [comment]
> _but I like **this** a lot. @twiro Please send a PR!_

The question is... what do you mean with "_this_"? ;)

The [quick and dirty hack](https://github.com/symphonycms/symphony-2/issues/2650#issuecomment-286672035) I proposed initially (bringing the button to front via z-index and avoid ugly overlappings) or one of the solutions that came up in the later discussion?

Actually I really like [Moritz' last proposal](https://github.com/symphonycms/symphony-2/issues/2650#issuecomment-286889299) and built a quick mockup visualizing how this could turn out:

![symphony-associations-drawer-4](https://cloud.githubusercontent.com/assets/870227/25610095/2ac1276c-2f22-11e7-8511-fea5506b1e53.jpg)

## Association Drawer - Proposal 2.7-A

Compared to the current state this would include the following **conceptual** changes:

1. Implementing two "main" groups in the drawer - one for listing sections/entries the current entry is "**Linked to**" and another for listing sections/entries the current entry has "**Links in**". Use that wording (which is already used now) for the two group headlines.
2. No more detailed wording for listing the linked sections - just the blank section-name followed by a number listing the amount of linked entries.
3. Reduce the content of the "Create New"-Buttons to a simple "+".

Regarding the **styling** I'd propose the following changes:

1. Use a bigger headline font size (same as for the main entry title/breadcrumb) for the grouping headlines to make sure they don't get lost if lots of sections/entries are listed.
2. Make the section names bold to improve the visual hierarchy and to emphasize their meaning as headlines for the entry-links listed below.
3. Reduce the line height for each linked entry a tiny bit.
4. Allow the drawer to become a little wider (at least on viewports that offer enough space) - I've used 320 pixels instead of 300 and this already makes quite a difference. The best solution in my eyes would be setting a minimum of 300 pixel (maybe up to a viewport width of 1440px), but letting the drawer grow dynamically on bigger screens. (Documenter would also benefit from this!). 

---

Here's another example how this would look with the german wording and a section that doesn't have linked entries yet. It also compares one mockup with the current wording for the buttons and one with a simple "+".

Though the initial issue would mostly be fixed if we kept the "full text buttons" I'd vote for the leaner "+"-solution. It looks cleaner and - in my opinion - is not only "enough" button (in giving it a little less relevance compared to the entry-links which i see as a good and apropriate shift in this context), but also not really harder to understand.

![symphony-associations-drawer-5](https://cloud.githubusercontent.com/assets/870227/25611171/633fd7b0-2f26-11e7-8682-e062f9e1bba4.jpg)

So if we can agree upon this I'll gladly try to implement it and send a PR!

What do you think?

**Comment by nitriques** [comment]
@twiro 

LOL, `this = Moritz' last proposal` (Haven't you learn Math?)

I often only have the time to read github threads late at night after my boy is sleeping, so I may not be always 100% clear. I will make typos and errors, sorry eh.

> What do you think?

I think it sounds great.  I like the clean separation between the 2 directions of the associations. But shouldn't we invert the arrow ?  --> linked to, <-- linked in ? cc @nilshoerrmann

**Comment by nilshoerrmann** [comment]
Your proposal is looking good, a few comments:

- The ellipsis in the title makes it look unfinished (in the sense of drafted). There should either be no punctuation or a colon in my eyes. I'd use now punctuation at all.
- In English the titles could just be "Links" and "Linked". (Deutsch ist immer etwas schwierig hier, aber ich könnte mir sogar eher eine Benennung wie "Eingehende Verknüpfungen" und "Ausgehende Verknüpfungen". Ja, ist wieder was Langes, ich weiß, aber die offenen Halbsätze überzeugen mich nicht).
- I agree with Nicolas that the arrows should reflect the association direction (incoming/outgoing)
- The `+` button looks a bit off in size and typography. This should either be text in my eyes, maybe shortened to `New` instead of `Create New`. Or it should be a propper graphical icon.

**Comment by nitriques** [comment]
>  This should either be text in my eyes, maybe shortened to New instead of Create New. Or it should be a propper graphical icon.

I also agree on this. Thanks!

**Comment by twiro** [comment]
Hey guys,

didn't manage to get this done before my holidays but will take care about this as soon as I'm back at the end of may - hopefully that's still in time for the upcoming 2.7 release.

Cheers, roman

**Comment by nitriques** [comment]
Thanks @twiro . I would not mind releasing 2.8.0 if it does not make it into 2.7.0. There's just numbers, not need to wait for X or Y ;)

**Comment by twiro** [comment]
Continued the discussion in the PR: https://github.com/symphonycms/symphony-2/pull/2698#issuecomment-307122360

**Comment by nitriques** [comment]
Fixed in 051e0c770a
